api/src/test/java/org/openmrs/api/PersonServiceTest.java

```diff
@@ -2145,9 +2157,54 @@ public void savePersonAttributeType_shouldThrowAnErrorWhenTryingToSavePersonAttr
 		pat.setDescription("New person attribute type");
 		ps.savePersonAttributeType(pat);
 	}
+
+
+	/**
+	 * @see PersonService#retirePersonAttributeType(PersonAttributeType, String)
+	 * @throws APIException
+	*/
+	@Test
+	@Verifies(method = "retirePersonAttributeType(PersonAttributeType, String)", value = "should retire person attribute type when retire reason is not null nor empty")
+	public void retirePersonAttributeType_shouldRetirePersonAttributeTypeWhenRetireReasonIsNotNullNorEmpty()
+	        throws Exception {
+		PersonService ps = Context.getPersonService();
+		PersonAttributeType pat = ps.getPersonAttributeType(1);
+		ps.retirePersonAttributeType(pat, "health");
+		Assert.assertTrue(pat.isRetired());
+		assertEquals(Context.getAuthenticatedUser(), pat.getRetiredBy());
+		assertEquals("health", pat.getRetireReason());
+		assertEquals(new Date(), pat.getDateRetired());
+	}
+
 	
 	/**
 	 * @see PersonService#retirePersonAttributeType(PersonAttributeType, String)
+	 * @throws APIException
+	*/
+	@Test(expected = APIException.class)
```

Comment:
please use junits expected exception rule:
you can read about it here https://github.com/junit-team/junit4/wiki/Rules
 
and also find examples in openmrs-core 
* https://github.com/openmrs/openmrs-core/blob/a9a1fcee01ff6a8d40af6440be014a3b208ecc52/api/src/test/java/org/openmrs/DurationTest.java#L29-L30
* https://github.com/openmrs/openmrs-core/blob/a9a1fcee01ff6a8d40af6440be014a3b208ecc52/api/src/test/java/org/openmrs/DurationTest.java#L101-L102

this applies to all the tests you add which test for exceptions